File: chapter-1/lists-arrays.py
def print_list(my_list):
    # get the total length of the list
    for i in range(0, len(my_list)):
        print(my_list[i], end=" ")  # print the value of the list at index[i]
    print()
    return


my_list = [1, 2, 3, 4]  # list (a collection type)
print_list(my_list)


def print_list(my_list):
    for i in my_list:
        print(i, end=" ")
    print()
    return


# This function will return a specified list of values from 0, to n-1.
def create_list(n):
    my_list = []  # empty list
    for i in range(0, n):
        my_list.append(i)  # appends values (i) to the empty list up to n-1
    return my_list


my_list = create_list(5)
print_list(my_list)


def max_list(my_list):
    max = my_list[0]
    for i in my_list:
        if i > max:
            max = i
    return max

File: chapter-1/ListsArrays.java
public class ListsArrays {
    public static void main (String[] parms) {
       int[] my_list; // an array of type integer
       my_list = new int[] {1,2,3,4};
       print_list(my_list);

       System.out.println("Refactored print_list method:");
       my_list = create_list(5);
       refactored_print_list(my_list);
    }

    public static void print_list(int[] my_list) {
        int i;

        for(i=0; i<my_list.length; i++) {
            System.out.println(my_list[i] + " ");
        }
        System.out.println();
    }

    public static int[] create_list(int n) {
        int[] list;
        int i;

        list = new int[n];
        for (i=0; i<n; i++) {
            list[i] = i;
        }
        return list;
    }

    public static void refactored_print_list(int[] list) {
        int i;
        for (i=0; i < list.length; i++) {
            System.out.println(list[i] + " ");
        }
        System.out.println();
    }
}